ff/serving/metadata/metadata.go

```diff
@@ -1103,6 +1215,11 @@ func (serv *MetadataServer) genericCreate(ctx context.Context, res Resource, ini
 	if err := serv.lookup.Set(id, res); err != nil {
 		return nil, err
 	}
+	if serv.needsJob(res) {
```

Comment:
should make this atomic so a resource isn't set without the job being set
resource interface should have a function HasJob, and it can have a SetJob so that the resource is set with the job atomically